packages/features/eventtypes/components/tabs/availability/EventAvailabilityTab.tsx

```diff
@@ -439,42 +438,41 @@ const TeamAvailability = ({
   );
 };
 
+const useCommonScheduleState = (initialScheduleId: number | null) => {
+  const { setValue } = useFormContext<FormValues>();
+  const [useHostSchedulesForTeamEvent, setUseHostSchedulesForTeamEvent] = useState(!initialScheduleId);
+  // Reset the main schedule
+  const clearMainSchedule = () => {
+    setValue("schedule", null, { shouldDirty: Boolean(initialScheduleId) });
+  };
+  // Toggle function
+  const toggleScheduleState = (checked: boolean) => {
+    const useHostSchedulesForTeamEvent = !checked;
+    setUseHostSchedulesForTeamEvent(useHostSchedulesForTeamEvent);
+    if (useHostSchedulesForTeamEvent) clearMainSchedule();
+  };
+  return {
+    useHostSchedulesForTeamEvent,
+    toggleScheduleState,
+  };
+};
+
 const UseCommonScheduleSettingsToggle = ({ eventType, ...rest }: EventTypeScheduleProps) => {
   const { t } = useLocale();
-  const { setValue, resetField, getValues, watch } = useFormContext<FormValues>();
-
-  const [useHostSchedulesForTeamEvent, setUseHostSchedulesForTeamEvent] = useState(
-    !Boolean(eventType.schedule)
-  );
-
-  const watchHosts = watch("hosts");
-
+  const { useHostSchedulesForTeamEvent, toggleScheduleState } = useCommonScheduleState(eventType.schedule);
   return (
     <>
       <SettingsToggle
         checked={!useHostSchedulesForTeamEvent}
-        onCheckedChange={(checked) => {
-          setUseHostSchedulesForTeamEvent(!checked);
-          if (checked) {
-            if (Boolean(eventType.schedule)) resetField("schedule");
-            getValues("hosts").map((_, index) => {
```

Comment:
This seems like a bad idea (on check, remove all host schedule Ids)

Reply:
Yeah, this makes sense